codice/ddf
# [2.11.x] DDF-3817 Add support for separate external/internal urls

#### What does this PR do?
Adds support for external host and port properties in DDF. 
This is useful when DDF is behind a proxy and the URL a user sees is different than the local url DDF uses.
I still need to update documentation on this and any feedback on where that should go and what it should be would be helpful!

⚠️ **THIS PR WILL BREAK PROJECTS THAT DEPEND ON DDF**
⚠️ **SHOULD THE INTERNAL PROPERTIES BE RENAMED?**

#### Who is reviewing it? 
@alexaabrd 
@SmithJosh 
@tbatie 
@rzwiefel 
@clockard 

#### Choose 2 committers to review/merge the PR. 
@brendan-hofmann
@lessarderic

#### How should this be tested?  
Configure a DDF with an internal host of localhost and an external host of the machine's name. Configure them to be on different ports.
Verify that when accessing action providers in Intrigue that the link is the external host and port. 
IDP may need to be disabled to test this as it doesn't work with the external host/port. 

Optionally, use docker and nginx to set up a DDF with a reverse proxy and verify things work as expected. 

#### Any background context you want to provide?
With DDF-2.11.6, when a DDF is set up with a reverse proxy, the action providers in Intrigue don't work as expected. 

#### What are the relevant tickets?
[DDF-3817](https://codice.atlassian.net/browse/DDF-3817)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
